### PR TITLE
Add hardware resource warning to server audit widget

### PR DESCRIPTION
**Pull request by rparth07**
- Implemented a new error code type `hardwarewarning` to indicate insufficient hardware resources.
- Updated server audit widget to display the new hardware warning message.
- Included link to hardware requirements documentation for user reference.

Change-Id: I3151b57acc1275564d8de1cdc18922685e8edebe

* Resolves: #9491
* Target version: master 

### Checklist

- [X] I have run `make prettier-write` and formatted the code.
- [X] All commits have Change-Id
- [ ] I have run tests with `make check`
- [X] I have issued `make run` and manually verified that everything looks okay
- [X] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: wsd/COOLWSD.hpp [hunk 1]
```diff
@@ -350,6 +351,11 @@ class COOLWSD final : public Poco::Util::ServerApplication,
         return Util::encodeId(NextConnectionId++, 3);
     }
 
+    static std::string getHardwareResourceWarning()
```
Comment (eszkadev) [inline wsd/COOLWSD.hpp:354]: `const std::string&` ?

Comment (rparth07) [reply wsd/COOLWSD.hpp]: Thank you for the suggestion. 
I have updated the getHardwareResourceWarning method to return a constant reference

**Comment by eszkadev** [comment]
```
Checking formatting...
[warn] src/control/Control.ServerAuditDialog.ts
[warn] Code style issues found in the above file. Run Prettier with --write to fix.
make[1]: *** [Makefile:1479: prettier] Error 1
```

please apply formatting: `make prettier-write` and commit changes

**Review by Rash419 — CHANGES_REQUESTED** [review]
Link should `https://sdk.collaboraonline.com/docs/installation/Configuration.html#performance`
Remove the highlight parameter please  :)

**Comment by Rash419** [comment]
Other note: https://sdk.collaboraonline.com/docs/installation/Configuration.html#performance doesn't contain recommendation for hardware so that this warning can be resolved. We should update our SDK with recommendation. But sdk changes are out of scope for this PR

cc: @pedropintosilva @Darshan-upadhyay1110

**Comment by rparth07** [comment]
Is there any appropriate link to resolve this warning?

**Comment by Darshan-upadhyay1110** [comment]
> Is there any appropriate link to resolve this warning?

@rparth07 appreciated but no worries for you on that part , we will update SDK Thanks :)
Thanks for the PR

**Comment by Darshan-upadhyay1110** [comment]
> Is there any appropriate link to resolve this warning?

Please change URL as per review comment and rebase your PR.

**Comment by pedropintosilva** [comment]
> Other note: https://sdk.collaboraonline.com/docs/installation/Configuration.html#performance doesn't contain recommendation for hardware so that this warning can be resolved. We should update our SDK with recommendation. But sdk changes are out of scope for this PR
> 
> cc: @pedropintosilva @Darshan-upadhyay1110

let's not block this PR because of that

**Comment by Rash419** [comment]
> > Other note: https://sdk.collaboraonline.com/docs/installation/Configuration.html#performance doesn't contain recommendation for hardware so that this warning can be resolved. We should update our SDK with recommendation. But sdk changes are out of scope for this PR
> > cc: @pedropintosilva @Darshan-upadhyay1110
> 
> let's not block this PR because of that

I am not blocking because of that. I am blocking because link here: https://github.com/CollaboraOnline/online/pull/9619/files#diff-4a741452e86f71bc3af5c5351d3dc0da70fb77563778bdceaec1ce4d24e6bf28R83 should be `https://sdk.collaboraonline.com/docs/installation/Configuration.html#performance`

**Comment by pedropintosilva** [comment]
@Rash419 yes yes I got it :) I just wanted to reiterate what you already stated "outside of this scope"

**Comment by rparth07** [comment]
Updated URL and rebased. Can you please check?

**Review by Rash419 — APPROVED** [review]
Thanks @rparth07 <3

**Pull request merged**